File: Solo/Puzzles/on-boarding/OnBoarding.java
import java.util.*;
import java.io.*;
import java.math.*;

class Player {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        // game loop
        while (true) {
            // Lecture des données d'entrée : deux ennemis et leurs distances
            String enemy1 = in.next();  // nom de l'ennemi 1
            int dist1 = in.nextInt();   // distance de l'ennemi 1
            String enemy2 = in.next();  // nom de l'ennemi 2
            int dist2 = in.nextInt();   // distance de l'ennemi 2

            // (Optionnel) Débug : affiche les informations sur la sortie d'erreur
            System.err.println("Enemy1: " + enemy1 + " (dist: " + dist1 + ")");
            System.err.println("Enemy2: " + enemy2 + " (dist: " + dist2 + ")");

            // Comparer les distances et afficher le nom de l'ennemi le plus proche
            if (dist1 < dist2) {
                System.out.println(enemy1);
            } else {
                System.out.println(enemy2);
            }
        }
    }
}

File: Solo/Puzzles/on-boarding/OnBoarding.php
<?php
// Boucle infinie pour gérer le jeu (game loop)
while (TRUE) {
    // Lecture des données pour le premier ennemi
    fscanf(STDIN, "%s", $enemy1); // Nom de l'ennemi 1
    fscanf(STDIN, "%d", $dist1);  // Distance de l'ennemi 1

    // Lecture des données pour le deuxième ennemi
    fscanf(STDIN, "%s", $enemy2); // Nom de l'ennemi 2
    fscanf(STDIN, "%d", $dist2);  // Distance de l'ennemi 2

    // Afficher les ennemis et leur distance
    error_log("Enemy1: $enemy1, Distance1: $dist1");
    error_log("Enemy2: $enemy2, Distance2: $dist2");

    // Comparer les distances pour trouver l'ennemi le plus proche
    if ($dist1 < $dist2) {
        // Si l'ennemi 1 est plus proche, on le cible
        $target = $enemy1;
    } else {
        // Sinon, on cible l'ennemi 2
        $target = $enemy2;
    }

    // Echo directement la variable cible pour tirer
    echo "$target\n";
}
?>
